libqtile/widget/notify.py

```diff
@@ -191,7 +191,14 @@ def cmd_invoke(self):
             self.invoke()
 
     def finalize(self):
-        asyncio.create_task(self._finalize())
+        # We may need some async calls as part of the finalize call
+        # We run this with `call_soon_threadsafe` as this waits for
+        # the job to finish before continuing. This is important as,
+        # if the config is just reloading, we need to finish deregistering
+        # the notification server before the new Notify widget instance
+        # registers and creates a new server.
+        self.qtile.call_soon_threadsafe(self._finalize)
```

Comment:
I'm a little surprised that this fixes it. Why/how does this ensure that the `self._finalize` coroutine is executed before config reload?

Reply:
I'm also surprised! From what I could tell when looking into this, this call is synchronous and so blocks until the coroutine has completed.

I tested with lots of log messages to check the sequencing and it does indeed seem to work.

Reply:
OK, TIL! Thanks.